tests/test_connection.py

```diff
@@ -62,6 +102,23 @@
         pass
 
 
+class fail_after:
```

Comment:
Should trio.fail_after etc. be usable as decorators?  It seems to be foiled only by the "RuntimeError: must be called from async context" check.  @njsmith